lib/events.js

```diff
@@ -437,9 +441,9 @@ EventEmitter.prototype.removeAllListeners =
         this.removeListener(type, listeners);
       } else if (listeners) {
         // LIFO order
-        do {
-          this.removeListener(type, listeners[listeners.length - 1]);
-        } while (listeners[0]);
+        for (i = listeners.length; i-- > 0;) {
```

Comment:
FWIW the same pattern is also used here https://github.com/nodejs/node/blob/678480e328c1fdd152973695e978e0f02b489770/benchmark/events/ee-add-remove.js#L19